lua/catppuccin/lib/compiler.lua

```diff
@@ -8,8 +8,9 @@ local fmt = string.format
 local function inspect(t)
 	local list = {}
 	for k, v in pairs(t) do
-		local q = type(v) == "string" and [["]] or ""
-		table.insert(list, fmt([[%s = %s%s%s]], k, q, tostring(v), q))
+		local tv = type(v)
+		local q = tv == "string" and [["]] or ""
```

Comment:
If the type is a table then we don't need to also check if it is a string for quoting. I would have the table insert in an if block and the current impl in the else.